contracts/cw-core/src/contract.rs

```diff
@@ -379,7 +379,10 @@ pub fn execute_remove_item(
         return Err(ContractError::Unauthorized {});
     }
 
-    ITEMS.remove(deps.storage, key.clone());
+    if ITEMS.has(deps.storage, key.clone()) {
```

Comment:
I'm new to Rust, so please advise if there's a better way to do this. All this cloning is really strange just to satisfy the compiler, it seems that both `ITEMS.has` and `ITEMS.remove` only need read access to `key`, so why can't we pass it by reference?

Reply:
I think unfortunately these clones are needed. The reason we need to clone is because the remove function takes an item of type `K` where `K` is the key being used in the map. If it took `&K` we could pass by reference, but the API tells us that the function wants to take ownership of the key. Under the hood I believe it does this because it serializes the input key into a vec.